### PR TITLE
Adding the Legion Pro 7 16IRX9H model

### PR DESCRIPTION
**Pull request by felipelalli**
Initially, I simply copied it from the 16IRX8H, but I am still having issues with WiFi and sound (to be resolved).

Ref.: https://psref.lenovo.com/Product/Legion/Legion_Pro_7_16IRX9H

###### Description of changes
Adding the Legion Pro 7 16IRX9H model, copy/paste from 16IRX8H.

###### Things done

- [X] Tested the changes in your own NixOS Configuration
- [X] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: lenovo/legion/16irx9h/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ../../../common/cpu/intel
+    ../../../common/gpu/nvidia/prime.nix
+    ../../../common/gpu/nvidia/ada-lovelace
+    ../../../common/pc/laptop
+    ../../../common/pc/laptop/ssd
+    ../../../common/hidpi.nix
+  ];
+
+  boot.initrd.kernelModules = ["nvidia"];
+  boot.extraModulePackages = [config.boot.kernelPackages.lenovo-legion-module config.boot.kernelPackages.nvidia_x11];
```
Comment (Mic92) [inline lenovo/legion/16irx9h/default.nix:16]: the legion module is needed, but the rest should be already covered by the nvidia module.

File: lenovo/legion/16irx9h/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  hardware = {
+    nvidia = {
+      modesetting.enable = lib.mkDefault true;
```
Comment (Mic92) [inline lenovo/legion/16irx9h/default.nix:20]: Already enabled.

```suggestion
```

File: lenovo/legion/16irx9h/default.nix (hunk 1 above)
Comment (Mic92) [reply lenovo/legion/16irx9h/default.nix:16]:
```suggestion
  boot.extraModulePackages = [config.boot.kernelPackages.lenovo-legion-module];
```

**Comment by felipelalli** [comment]
@Mic92 I just copied it from https://github.com/NixOS/nixos-hardware/blob/master/lenovo/legion/16irx8h/default.nix is that wrong too? Should I fix it too in this same PR?

**Comment by Mic92** [comment]
@felipelalli I noticed that we actually don't add nvidiea to `boot.initrd.kernelModule`, which makes me wonder do you see any difference in having it in initrd? I.e. refreshing or changing resolution during boot?

**Comment by felipelalli** [comment]
> @felipelalli I noticed that we actually don't add nvidea to `boot.initrd.kernelModule`, which makes me wonder do you see any difference in having it in initrd? I.e. refreshing or changing resolution during boot?

Sorry, but I couldn't understand your question. Apologies, I am a newbie. I just copied the setup of a laptop similar to mine.

**Comment by Mic92** [comment]
@felipelalli I am talking about this line:

```
  boot.initrd.kernelModules = ["nvidia"];
```

When it's added does it flicker less during boot up. This line will put your nvidia driver into initrd and hence will initialize the graphic card earlier during boot.
I believe not having it would make your graphic driver to be somewhat re-initialized once filesystem are mounted, resulting in a refreshing the screen, potentially changing the resolution

File: lenovo/legion/16irx9h/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/gpu/nvidia/prime.nix
+    ../../../common/gpu/nvidia/ada-lovelace
+    ../../../common/pc/laptop
+    ../../../common/pc/laptop/ssd
+    ../../../common/hidpi.nix
+  ];
+
+  boot.initrd.kernelModules = ["nvidia"];
```
Comment (Mic92) [inline lenovo/legion/16irx9h/default.nix:15]: I don't think we should this option unless, we know why we are doing this:


```suggestion
```

https://github.com/NixOS/nixos-hardware/pull/1144

Comment (felipelalli) [reply lenovo/legion/16irx9h/default.nix:15]: OK. But it need to be fixed as well on [16irx8h](https://github.com/NixOS/nixos-hardware/blob/master/lenovo/legion/16irx8h/default.nix), am I right?

**Comment by felipelalli** [comment]
@Mic92 I have applied your suggestions, but pay attention that the same need to be done in 16irx8h.

**Comment by felipelalli** [comment]
Replaced by #1227

**Pull request closed without merging**